jobs/build-arch.Jenkinsfile

```diff
@@ -374,9 +374,10 @@ lock(resource: "build-${params.STREAM}-${basearch}") {
         }
 
         stage('Archive') {
-            shwrap("""
-            cosa compress --compressor xz
-            """)
+            // lower to make sure we don't go over and account for overhead
+            pipeutils.withXzMemLimit(cosa_memory_request_mb - 256) {
+                shwrap("cosa compress --compressor xz")
+            }
```

Comment:
This is technically running on the remote. Should we still apply the limit? If we do still apply the limit should we call out the elephant in the room (`"Yes, this is running on the remote, but we limit anyway because ..."`).

Reply:
Oh yes, good catch. I dropped that hunk. I think it could make sense to apply it to the remote to be considerate to other workloads, but would require doing something like `cosa shell -- env XZ_DEFAULTS=... cosa compress`? And the limit itself doesn't have to match what we use for the pod request of course.